Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift

```diff
@@ -91,23 +91,25 @@ extension PlatformIntelligenceTextEffect {
     }
 
     func targetedPreview(for chunk: UITextEffectTextChunk) async -> UITargetedPreview {
-        guard let chunk = chunk as? UITextEffectTextChunkAdapter<Wrapped.Chunk> else {
-            fatalError("Failed to create a targeted preview: parameter was of unexpected type \(type(of: chunk)).")
+        if let chunk = chunk as? UIPonderingTextEffectTextChunkAdapter<Wrapped.Chunk> {
+            return chunk.preview
         }
 
-        guard let preview = await self.wrapped.textPreview(for: chunk.wrapped) else {
-            fatalError("Failed to create a targeted preview: unable to create a preview from the given chunk.")
+        if let chunk = chunk as? UIReplacementTextEffectTextChunkAdapter<Wrapped.Chunk> {
+            return chunk.source
         }
 
-        return preview
+        fatalError("Failed to create a targeted preview: parameter was of unexpected type \(type(of: chunk)).")
```

Comment:
If `fatalError` is a release assert, I'm not sure this is safe, since `TextIndicator` creation can fail?

Same question applies to other use of `fatalError` in this patch.

Reply:
actually, all the fatal errors are safe (there are however a few ! which are not safe, so I’ll fix those)